com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/mqtt_streams.py

```diff
@@ -0,0 +1,193 @@
+        with a server configured to require that form of authentication.
+    
+     password
+        Optional String.  The identifier to use when authenticating
+        with server configured to require that form of authentication.
+    
+     trustStore
+        Optional String. The pathname to a file containing the
+        public certificate of trusted MQTT servers.  If a relative path
+        is specified, the path is relative to the application directory.
+        Required when connecting to a MQTT server with an 
+        ssl:/... serverURI.
+    
+     trustStorePassword
+        Required String when trustStore is used.
+        The password needed to access the encrypted trustStore file.
+    
+     keyStore
+        Optional String. The pathname to a file containing the
+        MQTT client's public private key certificates.
+        If a relative path is specified, the path is relative to the
+        application directory. 
+        Required when an MQTT server is configured to use SSL client authentication.
+    
+     keyStorePassword
+        Required String when keyStore is used.
+        The password needed to access the encrypted keyStore file.
+    
+     messageQueueSize
+        [subscribe] Optional Integer. The size, in number
+        of messages, of the subscriber's internal receive buffer.  Received
+        messages are added to the buffer prior to being converted to a
+        stream tuple. The receiver blocks when the buffer is full.
+        The default is 50.
+    
+     retain
+        [publish] Optional Boolean. Indicates if messages should be
+        retained on the MQTT server.  Default is false.
+    
+     qos
+        Optional Integer. The default
+         MQTT quality of service used for message handling.
+        The default is 0.
+    
+     additional information at http://mqtt.org
+     and http://ibmstreams.github.io/streamsx.messaging
+    """
+
+    def __init__(self,topology,config) :
+        self.topology = topology
+        self.config = config.copy()
+        self.opCnt = 0
+        
+    def publish(self,pub_stream,topic) :
+        parms = self.config.copy() 
+        #del parms['messageQueueSize'] 
+        parms['topic'] = topic
+        parms['dataAttributeName'] = "string"
+        if (++self.opCnt > 1) :
+            # each op requires its own clientID
+            clientId =  parms['clientID']
+            if (clientId is not None and len(clientId) > 0) :
+                parms['clientID'] = clientId+"-"+str(id(self))+"-"+str(self.opCnt)
+        # convert pub_stream outputport schema from spl po to spl rstring type
+        forOp = pub_stream._map(streamsx.topology.functions.identity,schema.CommonSchema.String)                
+        op = self.topology.graph.addOperator(kind="com.ibm.streamsx.messaging.mqtt::MQTTSink")
+        op.addInputPort(outputPort=forOp.oport)
+        #remove toolkitDir param since MQTTSink fails if set
+        del op.params['toolkitDir']
```

Comment:
I got this from the java implementation.  I assumed it was required to ensure unique cliendId for multiple operators using the same MqttStream class.